### PR TITLE
Add a 404 error page

### PR DESCRIPTION
**Pull request by Vinnl**
# References: 
Jira: MNTOR-1167
Figma: https://www.figma.com/file/gvyyqS9egsGjsH0LUDqT1m/Breach-Resolution-Flow-Design?node-id=654%3A16492&t=NYtMRtj552xcpQeO-4

# Description

(Also adding @mansaj as a reviewer, because I'm not sure if/how I should add logging to this, with the removal of the `NotFoundError`.)

Adds a 404 error page. Note that I kept the header; this allowed me to avoid another branch for the layout, and I imagine it's still useful when you're on the wrong page. ~~I'll also verify with UX whether that's OK.~~ [UX is OK with this.](https://mozilla-hub.atlassian.net/browse/MNTOR-1167?focusedCommentId=648427)

I also didn't have ideas on how to vertically center (classic...) the error message without pushing the header and footer out of the page or reworking the entire site layout mechanisms, so open to suggestions there.

# Screenshot (if applicable)

image

Screen Shot 2023-02-14 at 15 09 33

# How to test

Visit a page that does not exist.

# Checklist (Definition of Done)
- [x] Localization strings (if needed) have been added.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [x] I've added or updated the relevant sections in readme and/or code comments
- [ ] I've added a unit test to test for potential regressions of this bug. - Not sure where the best place for this is.
- [ ] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [ ] All acceptance criteria are met. - This is only the 404 page, since feedback here probably applies to all error pages.
- [x] Jira ticket has been updated (if needed) to match changes made during the development process.
- [x] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

File: locales/en/app.ftl
Comment (flodolo) [inline locales/en/app.ftl:846]: Page not found is only 404. What's the benefit of using a variable here?

**Review by flodolo — APPROVED** [review]

File: src/middleware/error.js
Comment (mansaj) [inline src/middleware/error.js:34]: For logging, what if you kept the `notFound` function, and instead of throwing an error, just log the error in there and then renders the page?

**Review by mansaj — APPROVED** [review]

File: src/client/js/partials/notFound.js
Comment (flozia) [inline src/client/js/partials/notFound.js:1]: The license header is missing.

```suggestion
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 if (window.history.length > 1) {
```

File: src/client/css/partials/notFound.css
Comment (flozia) [inline src/client/css/partials/notFound.css:24]: “No newline at end of file”. :)

File: src/client/css/variables.css
Comment (flozia) [inline src/client/css/variables.css:33]: Suggestion: Identation

```suggestion
  --purple-60: #7542e5;
```

File: src/client/js/partials/notFound.js
Comment (flozia) [inline src/client/js/partials/notFound.js:15]: Question (non-blocking): Would it be possible to pass the prev path from the controller and append it to `#back-cta` via a `data-*` attribute?

**Review by flozia — APPROVED** [review]
The page is looking great! Approving with no blocking remarks, but some nits that should be addressed before merging.

File: src/client/js/partials/notFound.js
Comment (Vinnl) [reply src/client/js/partials/notFound.js:1]: Fixed in 2f5a84c9a417b67709f2dc007c429f1926b4aac3

BUT it's a waste of your and my time to manually be checking this in reviews, so I also submitted https://github.com/mozilla/blurts-server/pull/2836 as well.

File: src/client/css/variables.css
Comment (Vinnl) [reply src/client/css/variables.css:33]: Oh sorry, forgot to call this out - I cannot get VSCode to _not_ add this indentation back. I expect it's something in our shared VSCode settings?

Comment (Vinnl) [reply src/client/css/variables.css]: Actually, looks like it will let me now: af46b139

File: src/client/css/partials/notFound.css
Comment (Vinnl) [reply src/client/css/partials/notFound.css]: I think this was the same story of VSCode automatically removing it, but it works now: af46b139

File: src/client/js/partials/notFound.js
Comment (Vinnl) [reply src/client/js/partials/notFound.js:19]: Oof. Ehm, maybe, but then we'd be tracking the user's navigation state on the server-side, which I think is a bit out of scope for this PR...

File: src/middleware/error.js
Comment (Vinnl) [reply src/middleware/error.js:42]: Yeah that works for me. I'm not entirely sure about the format of our log messages though. What do you think about 500e23c5?

File: src/client/js/partials/notFound.js
Comment (flozia) [reply src/client/js/partials/notFound.js:19]: Totally out of scope — just left this as a thought in case we need it in the future.

File: src/client/js/partials/notFound.js
Comment (flozia) [reply src/client/js/partials/notFound.js:5]: Awesome — ty!

File: src/middleware/error.js
Comment (mansaj) [reply src/middleware/error.js:42]: Works for me

**Review by mansaj — APPROVED** [review]

File: src/views/main.js
Comment (toufali) [inline src/views/main.js:50]: With the addition of this error page we should probably create a separate layout to handle these two partials and the others to come that don't require UI for logged-in user.  I created the following task to track this:
https://mozilla-hub.atlassian.net/browse/MNTOR-1254

**Pull request merged**
